dev/io.openliberty.security.jakartasec.3.0.internal_fat/fat/src/io/openliberty/security/jakartasec/fat/tests/TokenValidationTests.java

```diff
@@ -0,0 +1,1219 @@
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_ISSUED_AT_TIME_IN_SECS);
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet", getMissingClaimExpectations(PayloadConstants.PAYLOAD_ISSUED_AT_TIME_IN_SECS));
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint with an iat claim value that is in the future will fail validation.
+     */
+    @Test
+    public void TokenValidationTests_tokens_iatInTheFuture() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_ISSUED_AT_TIME_IN_SECS);
+        builder.setIssuedAt(NumericDate.fromSeconds(1735689540)); // 12/31/2024 11:59:00 PM
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet", getIatInFutureExpectations(PayloadConstants.PAYLOAD_ISSUED_AT_TIME_IN_SECS));
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint with an iat claim value that valid (in the past) and will pass validation
+     */
+    @Test
+    public void TokenValidationTests_tokens_iatValid() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder();
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet");
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint without an nbf claim will pass validation and nbf is not required.
+     */
+    @Test
+    public void TokenValidationTests_tokens_nbfMissing() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_NOT_BEFORE_TIME_IN_SECS);
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet");
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint with an nbf claim that is in the future (but before the exp) will fail validation as the token is NOT valid yet.
+     */
+    @Test
+    public void TokenValidationTests_tokens_nbfInTheFuture() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder(PayloadConstants.PAYLOAD_EXPIRATION_TIME_IN_SECS);
+        builder.setExpirationTime(NumericDate.fromSeconds(1735693200));// 01/01/2025 01:00:00 AM
+        builder.setNotBefore(NumericDate.fromSeconds(1735689540)); // 12/31/2024 11:59:00 PM
+        saveTokenEndpointToken(builder);
+
+        runEndToEndTest("TokenTestAppServlet", getIatInFutureExpectations(PayloadConstants.PAYLOAD_NOT_BEFORE_TIME_IN_SECS));
+
+    }
+
+    /**
+     * Test that a token returned from the token endpoint with an nbf claim that is in the future (but after the exp) will fail validation as the exp can not be before nbf.
+     */
+    @Test
+    public void TokenValidationTests_tokens_nbfAfterExp() throws Exception {
+
+        JWTTokenBuilder builder = initTokenBuilder();
+        builder.setNotBefore(NumericDate.fromSeconds(1735689540)); // 12/31/2024 11:59:00 PM
```

Comment:
Set the value programmatically.

Reply:
updated - was originally going to set it out a century, but ...